### PR TITLE
[demux] WaitForSamples returns both live and timed metrics.

### PR DESCRIPTION
**Pull request by remeh**
### What does this PR do?

A unit tests improvement around the `WaitForSamples` method in the mocked `AgentDemultiplexer` which now returns both live metrics and timed metrics.

### Describe how to test/QA your changes

No QA needed: It is only modifying unit tests, if they pass, we're all good for the QA.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by vickenty — APPROVED** [review]

File: pkg/aggregator/demultiplexer_mock.go [hunk 1]
```diff
@@ -52,40 +52,43 @@ func (a *TestAgentDemultiplexer) AddLateMetrics(metrics metrics.MetricSampleBatc
 	a.Unlock()
 }
 
-func (a *TestAgentDemultiplexer) samples() []metrics.MetricSample {
+func (a *TestAgentDemultiplexer) samples() ([]metrics.MetricSample, []metrics.MetricSample) {
```
Comment (carlosroman) [inline pkg/aggregator/demultiplexer_mock.go:55]: All looks good, just a small suggestion (mainly as I'll forget which is which)
```suggestion
func (a *TestAgentDemultiplexer) samples() (received []metrics.MetricSample, late []metrics.MetricSample) {
```
Same for `WaitForSamples` on line 72.

Comment (carlosroman) [reply pkg/aggregator/demultiplexer_mock.go:55]: Or should that `(received []metrics.MetricSample, timed []metrics.MetricSample)`?

**Review by carlosroman — APPROVED** [review]

File: pkg/aggregator/demultiplexer_mock.go (hunk 1 above)
Comment (remeh) [reply pkg/aggregator/demultiplexer_mock.go:55]: Good catch, I'll used `timed`.

Note that after a discussion with Maxime I also have a ticket around homogenizing this for TimedMetric or TimedSamples in the API and such, but it is opening a few questions I have to solve (TimedSample is way similar-looking to TimeSampler).

**Review by maxday — APPROVED** [review]
LGTM!

**Pull request merged**